go/private/extensions.bzl

```diff
@@ -15,40 +15,130 @@ _download_tag = tag_class(
 
 _host_tag = tag_class(
     attrs = {
-        "name": attr.string(mandatory = True),
+        "name": attr.string(),
         "version": attr.string(),
     },
 )
 
+# This limit can be increased essentially arbitrarily, but doing so will cause a rebuild of all
+# targets using any of these toolchains due to the changed repository name.
+_MAX_NUM_TOOLCHAINS = 9999
+_TOOLCHAIN_INDEX_PAD_LENGTH = len(str(_MAX_NUM_TOOLCHAINS))
+
 def _go_sdk_impl(ctx):
+    multi_version_module = {}
+    for module in ctx.modules:
+        if module.name in multi_version_module:
+            multi_version_module[module.name] = True
+        else:
+            multi_version_module[module.name] = False
+
+    toolchains = []
     for module in ctx.modules:
-        for download_tag in module.tags.download:
+        for index, download_tag in enumerate(module.tags.download):
             # SDKs without an explicit version are fetched even when not selected by toolchain
             # resolution. This is acceptable if brought in by the root module, but transitive
             # dependencies should not slow down the build in this way.
             if not module.is_root and not download_tag.version:
                 fail("go_sdk.download: version must be specified in non-root module " + module.name)
-            go_download_sdk(
-                name = download_tag.name,
+
+            # SDKs with an explicit name are at risk of colliding with those from other modules.
+            # This is acceptable if brought in by the root module as the user is responsible for any
+            # conflicts that arise. rules_go itself provides "go_default_sdk", which is used by
+            # Gazelle to bootstrap itself.
+            # TODO: Investigate whether Gazelle can use the first user-defined SDK instead to
```

Comment:
Might be worth creating an issue in gazelle to track this.

Reply:
Created https://github.com/bazelbuild/bazel-gazelle/issues/1469 and referenced it in the comment.